### PR TITLE
Added Private Auction Functionality

### PR DESCRIPTION
**Pull request by shakti97**
It will resolve and close issue #10

### CODE REVIEW

File: assets/app/jsx/AdminPanel.jsx [hunk 1]
```diff
@@ -295,18 +301,8 @@ class AdminPanel extends Component {
...
-            max_users,
-            is_open,
-            can_register,
-            clientIds,
-            catalogs,
-            sold,
-            start
-        } = this.state;
+        const {activeUsers,
+            q_type, clientIds,catalogs, sold, start, url_slug, max_users, is_open, can_register, access_type, password} = this.state;
```
Comment (adarshPatel509) [inline assets/app/jsx/AdminPanel.jsx]: Format it properly. do "$npm run fmt".

File: assets/app/jsx/AdminPanel.jsx [hunk 2]
```diff
@@ -315,18 +311,21 @@ class AdminPanel extends Component {
...
+                                access_type,
+                                password
                             }}
                             validate={values => {
                                 const errors = {};
                                 if (!values.url_slug) {
                                     errors.url_slug = 'Required';
-                                }
-                                if (!values.max_users) {
+                                } else if (!values.max_users) {
```
Comment (adarshPatel509) [inline assets/app/jsx/AdminPanel.jsx:346]: also add check for max_users <= 0.

File: assets/app/jsx/Home.jsx [hunk 3]
```diff
@@ -44,6 +45,45 @@ class Home extends Component {
...
+                                text: err
+                            });
+                        });
+                },
+                allowOutsideClick: () => !Swal.isLoading()
+            });
+        } else {
+            this.props.history.push('/auction/' + auction.auction_url);
+        }
+    };
```
Comment (adarshPatel509) [inline assets/app/jsx/Home.jsx]: user can directly goto /auction/slug and will be able to join a protected auction. So, have this check on auctionPage  instead, and create a socket connection only after  auctionAuth success.

File: models/auctionconfig.js [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
-      allowNull: false,
+      allowNull: false
+    },
+    access_type: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false
```
Comment (adarshPatel509) [inline models/auctionconfig.js]: add this in db_migration file also.

File: src/routes/auctionConfig.js [hunk 5]
```diff
@@ -25,7 +26,17 @@ app.use(
 app.use(bodyParser.json());
 
 app.post('/auctionConfig', function(req, res) {
-    const {q_type, user_id, can_register, is_open, url_slug: auction_url, max_users, owner_id} = req.body;
+    const {
+        q_type,
```
Comment (adarshPatel509) [inline src/routes/auctionConfig.js]: remove q_type and write separate functions for addConfig, getConfig and updateConfig.

**Review by adarshPatel509 — CHANGES_REQUESTED** [review]

File: assets/app/jsx/Auction.jsx [hunk 6]
```diff
@@ -36,8 +37,66 @@ class Auction extends Component {
...
         this.checkStatus();
     }
-
     componentDidMount() {
+        console.log('componentDid Mount');
+        this.isAuthenticated();
+    }
+    isAuthenticated = () => {
+        const {namespace} = this.state;
+        console.log('namespace ', namespace);
```
Comment (adarshPatel509) [inline assets/app/jsx/Auction.jsx]: remove all debug logs.

File: assets/app/jsx/Home.jsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import React, {Component} from 'react';
 import dataFetch from './DataFetch';
 import {notifyError} from '../Common/common';
+import Swal from 'sweetalert2';
```
Comment (adarshPatel509) [inline assets/app/jsx/Home.jsx]: remove this import. its not used anywhere.

File: app.js [hunk 8]
```diff
@@ -63,6 +63,7 @@ io.sockets.on('connection', socket => {
         handleConnections.currentCatalog(socket, namespace, owner_id, catalog);
     });
     socket.on('biddingStop', (owner_id, namespace) => {
+        // console.log('bidding Stop');
```
Comment (hariprasanths) [inline app.js]: remove all unncessary logs

File: migrations/20191107202814-create-user.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
-'use strict';
```
Comment (hariprasanths) [inline migrations/20191107202814-create-user.js:1]: migrations and models needs to be formatted

**Review by adarshPatel509 — APPROVED** [review]

**Review by hariprasanths — CHANGES_REQUESTED** [review]
Reloading the admin panel changes the auction status from private to public

File: migrations/20191107202814-create-user.js (hunk 9 above)
Comment (shakti2379) [reply migrations/20191107202814-create-user.js:1]: done

File: app.js (hunk 8 above)
Comment (shakti2379) [reply app.js]: done

File: assets/app/jsx/Auction.jsx [hunk 10]
```diff
@@ -36,8 +37,61 @@ class Auction extends Component {
     componentWillMount() {
         this.checkStatus();
```
Comment (hariprasanths) [inline assets/app/jsx/Auction.jsx]: Move this to DidMount()

File: models/auctionconfig.js [hunk 11]
```diff
@@ -1,35 +1,43 @@
-'use strict';
```
Comment (hariprasanths) [inline models/auctionconfig.js:1]: Format this file

File: app.js [hunk 12]
```diff
@@ -66,6 +66,7 @@ io.sockets.on('connection', socket => {
         handleConnections.stopBidding(io, socket, namespace, owner_id);
     });
     socket.on('disconnect', reason => {
+        console.log('disconnect', reason, socket);
```
Comment (hariprasanths) [inline app.js]: Remove this

File: assets/app/jsx/Auction.jsx (hunk 10 above)
Comment (shakti2379) [reply assets/app/jsx/Auction.jsx]: Done

File: models/auctionconfig.js (hunk 11 above)
Comment (shakti2379) [reply models/auctionconfig.js:1]: Done

File: app.js (hunk 12 above)
Comment (shakti2379) [reply app.js]: Done

**Review by hariprasanths — APPROVED** [review]

**Pull request merged**